library/core/src/panicking.rs

```diff
@@ -133,35 +138,70 @@ pub fn assert_matches_failed<T: fmt::Debug + ?Sized>(
             fmt::Display::fmt(self.0, f)
         }
     }
-    assert_failed_inner(AssertKind::Match, &left, &Pattern(right), args);
+    assert_failed_inner(
+        AssertKind::Match,
+        left_expr,
+        &left_val,
+        right_pattern,
+        &Pattern(right_pattern),
+        args,
+    );
 }
 
 /// Non-generic version of the above functions, to avoid code bloat.
 #[track_caller]
 fn assert_failed_inner(
     kind: AssertKind,
-    left: &dyn fmt::Debug,
-    right: &dyn fmt::Debug,
+    left_expr: &'static str,
+    left_val: &dyn fmt::Debug,
+    right_expr: &'static str,
+    right_val: &dyn fmt::Debug,
     args: Option<fmt::Arguments<'_>>,
 ) -> ! {
-    let op = match kind {
-        AssertKind::Eq => "==",
-        AssertKind::Ne => "!=",
-        AssertKind::Match => "matches",
+    let (op, macro_name) = match kind {
+        AssertKind::Eq => ("==", "assert_eq"),
+        AssertKind::Ne => ("!=", "assert_ne"),
+        AssertKind::Match => ("matches", "assert_matches"),
     };
 
     match args {
-        Some(args) => panic!(
-            r#"assertion failed: `(left {} right)`
+        Some(args) => panic_fmt(
+            format_args!(
+                r#"assertion failed: `(left {} right)`
   left: `{:?}`,
  right: `{:?}`: {}"#,
-            op, left, right, args
+                op, left_val, right_val, args
+            ),
+            Some(ExtraInfo::AssertInfo(&AssertInfo {
+                assertion: Assertion::Binary(BinaryAssertion {
+                    op,
+                    left_expr,
+                    right_expr,
+                    left_val,
+                    right_val,
+                }),
+                macro_name,
+                message: Some(args),
+            })),
         ),
-        None => panic!(
-            r#"assertion failed: `(left {} right)`
+        None => panic_fmt(
```

Comment:
`assert_failed_inner` is now implemented with `panic_fmt` instead of `panic!`. Is this change ok?

I use `panic_fmt` because I need to pass the `ExtraInfo`, but the implementation of `panic!` depends on the edition. I don't know what that means for the `assert_eq` and `assert_ne` macros.